lib/vector-core/src/event/log_event.rs

```diff
@@ -697,42 +729,44 @@ impl From<&tracing::Event<'_>> for LogEvent {
                 Value::Null
             },
         );
-        log.insert("metadata.level", meta.level().to_string());
+        log.insert(&TRACING_TARGET_PATHS.level, meta.level().to_string());
         log.insert(
-            "metadata.module_path",
+            &TRACING_TARGET_PATHS.module_path,
             meta.module_path()
                 .map_or(Value::Null, |mp| Value::Bytes(mp.to_string().into())),
         );
-        log.insert("metadata.target", meta.target().to_string());
-
+        log.insert(&TRACING_TARGET_PATHS.target, meta.target().to_string());
         log
     }
 }
 
 impl tracing::field::Visit for LogEvent {
     fn record_str(&mut self, field: &tracing::field::Field, value: &str) {
-        self.insert(field.name(), value.to_string());
+        self.parse_path_and_insert(field.as_ref(), value.to_string())
+            .ok();
```

Comment:
I'm wondering if this is necessary. 

I think this is being used by the `internal_logs` source to convert a tracing log message into a Vector log message. If that is the case, to my knowledge, these fields will never be nested fields. If Vector were to emit a log with a field "blah.umm" that should be considered as a single field rather than a nested one. I'm not sure if that happens though...

@neuronull do you have an idea?

Reply:
Note: When I used `event_path!` vs `parse_path_and_insert` nothing changed/failed. This demonstrates a lack of test coverage.

cc @fuchsnj (in case you have any context on this)

Reply:
This is exactly why we are trying to move away from using strings as paths, because it doesn't tell you _how_ they should be parsed (if it should be nested or not). The real problem here is that the function accepts `fields: &[impl AsRef<str>]`. 
If this is getting fields from an existing Event, it should probably be passing in already parsed paths.